### PR TITLE
#2172 - update hermes json files

### LINKED ISSUE #2172: Update Hermes json test files
## Background
Hermes json files are almost two years old and not used for testing.

## Feature
Some of information were changed in time.
Found changes:
- version of spark-job library
- rename of credentials file
- rename of war file - path to it
- some tests are json not csv based

Found problem with spark 3:
- new config parameter need to be add to control usage of 'datetimeRebaseMode' for time data older than 1900
- Hermes logic for this logic has been implemented in [here](https://github.com/AbsaOSS/hermes/issues/131).

### PR DESCRIPTION
**Pull request by miroslavpojer**
- updated json files
- repair of typo in menas password configuration

New hermes json files can be tested with this [Hermes PR ](https://github.com/AbsaOSS/hermes/issues/131)

Closes #2172

### CODE REVIEW

File: examples/data/e2e_tests/test_jsons/castingConformanceRuleTest.json [hunk 1]
```diff
@@ -2,9 +2,9 @@
 {
     "vars": {
       "spark-submit": "spark-submit --num-executors 2 --executor-memory 2G --deploy-mode client",
-      "spark-conf": "--conf 'spark.driver.extraJavaOptions=-Denceladus.rest.uri=http://localhost:8080/rest_api/api -Dspline.mongodb.name=spline -Dspline.mongodb.url=mongodb://127.0.0.1:27017/ -Denceladus.recordId.generation.strategy=stableHashId'",
-      "enceladus-job-jar": "spark-jobs/target/spark-jobs-2.19.0-SNAPSHOT.jar",
-      "credentials": "--rest-api-credentials-file ~/.ssh/menasCredential.properties",
+      "spark-conf": "--conf 'spark.driver.extraJavaOptions=-Denceladus.rest.uri=http://localhost:8080/rest_api_war/api -Dspline.mongodb.name=spline -Dspline.mongodb.url=mongodb://127.0.0.1:27017/ -Denceladus.recordId.generation.strategy=stableHashId'",
```
Comment (benedeki) [inline examples/data/e2e_tests/test_jsons/castingConformanceRuleTest.json:5]: I think `rest-api` is also in the default resource files. Shouldn't it be consistent? 🤔

File: examples/data/e2e_tests/test_jsons/castingConformanceRuleTest.json [hunk 2]
```diff
@@ -2,9 +2,9 @@
...
+      "enceladus-job-jar": "spark-jobs/target/spark-jobs-3.0.0-SNAPSHOT.jar",
+      "credentials": "--rest-api-credentials-file ~/.ssh/menas-credential.properties",
```
Comment (benedeki) [inline examples/data/e2e_tests/test_jsons/castingConformanceRuleTest.json:7]: Why this change? Seems like personal setup.

File: examples/data/e2e_tests/test_jsons/coalesceConformanceRuleTest.json [hunk 3]
```diff
@@ -2,9 +2,9 @@
 {
     "vars": {
       "spark-submit": "spark-submit --num-executors 2 --executor-memory 2G --deploy-mode client",
-      "spark-conf": "--conf 'spark.driver.extraJavaOptions=-Denceladus.rest.uri=http://localhost:8080/rest_api/api -Dspline.mongodb.name=spline -Dspline.mongodb.url=mongodb://127.0.0.1:27017/ -Denceladus.recordId.generation.strategy=stableHashId'",
-      "enceladus-job-jar": "spark-jobs/target/spark-jobs-2.19.0-SNAPSHOT.jar",
-      "credentials": "--rest-api-credentials-file ~/.ssh/menasCredential.properties",
+      "spark-conf": "--conf 'spark.driver.extraJavaOptions=-Denceladus.rest.uri=http://localhost:8080/rest_api_war/api -Dspline.mongodb.name=spline -Dspline.mongodb.url=mongodb://127.0.0.1:27017/ -Denceladus.recordId.generation.strategy=stableHashId'",
+      "enceladus-job-jar": "spark-jobs/target/spark-jobs-3.0.0-SNAPSHOT.jar",
```
Comment (benedeki) [inline examples/data/e2e_tests/test_jsons/coalesceConformanceRuleTest.json:6]: Wonder if this somehow could be integrated with the current version. 🤔 
This way it will be behind next month.

Comment (miroslavpojer) [reply examples/data/e2e_tests/test_jsons/coalesceConformanceRuleTest.json:6]: Added as point for improvement in Issue #2168

File: examples/data/e2e_tests/test_jsons/castingConformanceRuleTest.json (hunk 2 above)
Comment (miroslavpojer) [reply examples/data/e2e_tests/test_jsons/castingConformanceRuleTest.json:7]: I would like to see it as small improvement to reach same pattern where CamelCase is used for Class files only. I am used to write these non-class files in format "word-word" or "word_word". I can see more similar locations with this naming. Are we following some format rules when to use CamelCase, w-w and w_w?

File: examples/data/e2e_tests/test_jsons/castingConformanceRuleTest.json (hunk 1 above)
Comment (miroslavpojer) [reply examples/data/e2e_tests/test_jsons/castingConformanceRuleTest.json]: Sure. "_war" removed from all files in commit 314a6b24073cae4a60931ed9885c319157142c84.

**Review by benedeki — DISMISSED** [review]
- [x] code reviewed
- [ ] pulled
- [ ] built
- [ ] run

**Review by Zejnilovic — APPROVED** [review]

File: rest-api/src/test/resources/application.properties [hunk 4]
```diff
@@ -57,7 +57,7 @@ enceladus.rest.hadoop.auth.krb5.username=hdfs@EXAMPLE.COM
 enceladus.rest.hadoop.auth.krb5.keytab=hdfs.keytab
 
 enceladus.rest.auth.inmemory.user=user
-enceladus.rest.auth.inmemory.password=chang< )eme
+enceladus.rest.auth.inmemory.password=changeme
```
Comment (dk1844) [inline rest-api/src/test/resources/application.properties:60]: Are you sure this was a typo? 
Could it have been a way to force the user of hermes to provide their own `application.properties` while this file serving only as an example?

**Review by dk1844 — APPROVED** [review]
@benedeki is right about the hardcoded version problem.

Otherwise LGTM (just read the code)

File: rest-api/src/test/resources/application.properties (hunk 4 above)
Comment (miroslavpojer) [reply rest-api/src/test/resources/application.properties]: You are right. After some searching I have found answer from author about goal of this "typo". Reverted in commit 0f212739bf7dfdac1b9f24004149387a52bd66a1.

**Review by dk1844 — DISMISSED** [review]
LGTM (just read the code)

File: examples/data/e2e_tests/test_jsons/stdNtDnTest.json [hunk 5]
```diff
@@ -30,7 +30,7 @@
         "pluginName" : "DatasetComparison",
         "name": "DatasetComparison",
         "order" : 1,
-        "args" : ["--format", "parquet", "--new-path", "#{new-std-data-path}#", "--ref-path", "#{ref-std-data-path}#", "--keys", "name" ],
+        "args" : ["--format", "parquet", "--new-path", "#{new-std-data-path}#", "--ref-path", "#{ref-std-data-path}#", "--keys", "property", "--datetimeRebaseMode", "LEGACY"],
```
Comment (Zejnilovic) [inline examples/data/e2e_tests/test_jsons/stdNtDnTest.json:33]:
```suggestion
        "args" : ["--format", "parquet", "--new-path", "#{new-std-data-path}#", "--ref-path", "#{ref-std-data-path}#", "--keys", "property"],
```

I think we said this is not needed anymore

Comment (Zejnilovic) [reply examples/data/e2e_tests/test_jsons/stdNtDnTest.json:33]: So we updated Hermes to include `datetimeRebaseMode`? 

We removed the opting afaik - https://github.com/AbsaOSS/hermes/pull/132

Comment (miroslavpojer) [reply examples/data/e2e_tests/test_jsons/stdNtDnTest.json]: You are right, I was confused. `datetimeRebaseMode` removed in commit 92d983d65f9275664edbe8c3f8124b0ccd0b9561.

**Review by benedeki — APPROVED** [review]
My comments were addressed, seems the ones of others too...

**Review by dk1844 — DISMISSED** [review]
LGTM (just read the code)

**Review by dk1844 — APPROVED** [review]
LGTM (I am sorry you had to switch Enceladus version back and forth -- not my intention indeed to complicate related setup)
